### PR TITLE
 Fix https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=64110 

### PR DESCRIPTION
**Pull request by sashashura**

### CODE REVIEW

File: Packet++/src/IPv4Layer.cpp [hunk 1]
```diff
@@ -290,9 +290,15 @@ void IPv4Layer::parseNextLayer()
...
+		{
+			if (GREv0Layer::isDataValid(payload, payloadLen))
+				m_NextLayer = new GREv0Layer(payload, payloadLen, this, m_Packet);
+		}
 		else if (greVer == GREv1)
-			m_NextLayer = new GREv1Layer(payload, payloadLen, this, m_Packet);
+		{
+			if (GREv1Layer::isDataValid(payload, payloadLen))
+				m_NextLayer = new GREv1Layer(payload, payloadLen, this, m_Packet);
+		}
```
Comment (seladb) [inline Packet++/src/IPv4Layer.cpp:301]: It's a bit nicer to do:
```cpp
if (greVer == GREv0 && GREv0Layer::isDataValid(payload, payloadLen))
	m_NextLayer = new GREv0Layer(payload, payloadLen, this, m_Packet);
else if (greVer == GREv1 && GREv1Layer::isDataValid(payload, payloadLen))
	m_NextLayer = new GREv1Layer(payload, payloadLen, this, m_Packet);
else
	m_NextLayer = new PayloadLayer(payload, payloadLen, this, m_Packet);
```

File: Packet++/src/IPv6Layer.cpp [hunk 2]
```diff
@@ -242,9 +242,15 @@ void IPv6Layer::parseNextLayer()
...
+		{
+			if (GREv0Layer::isDataValid(payload, payloadLen))
+				m_NextLayer = new GREv0Layer(payload, payloadLen, this, m_Packet);
+		}
 		else if (greVer == GREv1)
-			m_NextLayer = new GREv1Layer(payload, payloadLen, this, m_Packet);
+		{
+			if (GREv1Layer::isDataValid(payload, payloadLen))
+				m_NextLayer = new GREv1Layer(payload, payloadLen, this, m_Packet);
+		}
```
Comment (seladb) [inline Packet++/src/IPv6Layer.cpp:253]: same here

File: Packet++/src/IPv4Layer.cpp (hunk 1 above)
Comment (sashashura) [reply Packet++/src/IPv4Layer.cpp:301]: But it means if it is `GREv0` and invalid, it will go to `new PayloadLayer`.

Comment (seladb) [reply Packet++/src/IPv4Layer.cpp:301]: yes, we always want to have a next layer... the default is `PayloadLayer` if there is no better option...

**Review by seladb — APPROVED** [review]

**Pull request merged**